pennylane/ops/qubit/state_preparation.py

```diff
@@ -176,6 +172,43 @@ def compute_decomposition(state, wires):
         """
         return [MottonenStatePreparation(state, wires)]
 
+    def state_vector(self, wire_order=None):
+        num_op_wires = len(self.wires)
+        param = math.cast(self.parameters[0], np.complex128)
```

Comment:
I think you need the precision part of the functions arguments because user can requester their own c_dtype in the device and it will not always be complex 128.

Reply:
I wasn't entirely sure how to handle this, because it's not tied to a device anymore. `math.cast` seemed to be the best way that also works with all interfaces. I didn't want to force any dtype here, nor did I want to force users to manually specify floats or complex values (why shouldn't `[1,0,0,0]` be a valid 2-qubit state?), but the `norm` check below requires it and that's why I added it. @albi3ro perhaps we should expand the `state_vector` function prototype to include `dtype` as a kwarg with `np.complex128` as the default (it dispatches to other interfaces properly)

Reply:
Yes, we should be able to specify the `dtype` for a simulation.

Reply:
kk. fortunately, adding an `__init__` to this class will mean I don't have to do this casting when computing `state_vector` so it's less of a concern in this PR (you can set dtype by setting the dtype of the parameter you provide), but in a new PR I'll add `dtype` to `state_vector`.